packages/loader/container-loader/src/container.ts

```diff
@@ -1359,11 +1373,37 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         protocol.quorum.on("addMember", (clientId, details) => {
             // This is the only one that requires the pending client ID
             if (clientId === this.pendingClientId) {
-                this.setConnectionState(ConnectionState.Connected);
+                let startTime: number | undefined;
+                if (this.prevClientLeftP?.isCompleted === false) {
+                    startTime = performance.now();
+                }
+                // Wait for previous client to leave the quorum before firing "connected" event.
+                if (this.prevClientLeftP) {
+                    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                    this.prevClientLeftP.promise.then((leaveReceived: boolean) => {
+                        this.logger.sendTelemetryEvent({
+                            eventName: "ConnectedAfterWait",
+                            duration: startTime !== undefined ? performance.now() - startTime : 0,
+                            timeout: !leaveReceived,
+                            outstandingOps: this._deltaManager.clientSequenceNumber -
+                                this._deltaManager.clientSequenceNumberObserved,
+                        });
+                        this.setConnectionState(ConnectionState.Connected);
+                    });
+                } else {
+                    this.setConnectionState(ConnectionState.Connected);
+                }
+            }
+        });
+
+        protocol.quorum.on("removeMember", (clientId) => {
+            // If the client which has left was us, then resolve the def. promise.
+            if (this.clientId === clientId && this.prevClientLeftP?.isCompleted === false) {
```

Comment:
Same as in another place - no need to check for this.prevClientLeftP?.isCompleted, just resolve promise